py/desispec/test/test_ql_qa.py

```diff
@@ -286,19 +284,197 @@ def testGetRMS(self):
+        hdr['DATE-OBS'] = '2018-09-23T08:17:03.988'
+
+        #- Dimensions per amp
+        ny = self.ny = 1800
+        nx = self.nx = 1800
+        noverscan = nover = 50
+
+        hdr['BIASSEC1'] = xy2hdr(np.s_[0:ny, nx:nx+nover])
+        hdr['DATASEC1'] = xy2hdr(np.s_[0:ny, 0:nx])
+        hdr['CCDSEC1'] = xy2hdr(np.s_[0:ny, 0:nx])
+
+        hdr['BIASSEC2'] = xy2hdr(np.s_[0:ny, nx+nover:nx+2*nover])
+        hdr['DATASEC2'] = xy2hdr(np.s_[0:ny, nx+2*nover:nx+2*nover+nx])
+        hdr['CCDSEC2'] =  xy2hdr(np.s_[0:ny, nx:nx+nx])
+
+        hdr['BIASSEC3'] = xy2hdr(np.s_[ny:ny+ny, nx:nx+nover])
+        hdr['DATASEC3'] = xy2hdr(np.s_[ny:ny+ny, 0:nx])
+        hdr['CCDSEC3'] = xy2hdr(np.s_[ny:ny+ny, 0:nx])
+
+        hdr['BIASSEC4'] = xy2hdr(np.s_[ny:ny+ny, nx+nover:nx+2*nover])
+        hdr['DATASEC4'] = xy2hdr(np.s_[ny:ny+ny, nx+2*nover:nx+2*nover+nx])
+        hdr['CCDSEC4'] =  xy2hdr(np.s_[ny:ny+ny, nx:nx+nx])
 
+        hdr['NIGHT'] = '20180923'
+        hdr['EXPID'] = 1
+        hdr['FLAVOR']='dark'
+
+        rawimage = np.zeros((2*ny, 2*nx+2*noverscan))
+        offset = {'1':100.0, '2':100.5, '3':50.3, '4':200.4}
+        gain = {'1':1.0, '2':1.5, '3':0.8, '4':1.2}
+        rdnoise = {'1':2.0, '2':2.2, '3':2.4, '4':2.6}
+
+        quad = {
+            '1': np.s_[0:ny, 0:nx], '2': np.s_[0:ny, nx:nx+nx],
+            '3': np.s_[ny:ny+ny, 0:nx], '4': np.s_[ny:ny+ny, nx:nx+nx],
+        }
+
+        for amp in ('1', '2', '3', '4'):
+
+            hdr['GAIN'+amp] = gain[amp]
+            hdr['RDNOISE'+amp] = rdnoise[amp]
+
+            xy = _parse_sec_keyword(hdr['BIASSEC'+amp])
+            shape = [xy[0].stop-xy[0].start, xy[1].stop-xy[1].start]
+            rawimage[xy] += offset[amp]
+            rawimage[xy] += np.random.normal(scale=rdnoise[amp], size=shape)/gain[amp]
+            xy = _parse_sec_keyword(hdr['DATASEC'+amp])
+            shape = [xy[0].stop-xy[0].start, xy[1].stop-xy[1].start]
+            rawimage[xy] += offset[amp]
+            rawimage[xy] += np.random.normal(scale=rdnoise[amp], size=shape)/gain[amp]
+
+        #- set CCD parameters
+        self.ccdsec1=hdr["CCDSEC1"]
+        self.ccdsec2=hdr["CCDSEC2"]
+        self.ccdsec3=hdr["CCDSEC3"]
+        self.ccdsec4=hdr["CCDSEC4"]
+
+        #- raw data are integers, not floats
+        rawimg = rawimage.astype(np.int32)
+        self.expid=hdr["EXPID"]
+        self.camera=hdr["CAMERA"]
+
+        #- read psf, should use specter.PSF.load_psf instead of desispec.PSF(), otherwise need to create a psfboot somewhere.
+
+        psf=self.psf
+
+        #- make the test pixfile, fibermap file
+        img_pix = rawimg
+        img_ivar = np.ones_like(img_pix) / 3.0**2
+        img_mask = np.zeros(img_pix.shape, dtype=np.uint32)
+        img_mask[200] = 1
+
+        #- manually insert sky peaks for xwsigma
+        zpeaks = np.array([8401.5,8432.4,8467.5,9479.4,9505.6,9521.8])
+        fibers = np.arange(5)
+        peak1 = PSF.xy(psf,fibers,zpeaks[0])
```

Comment:
This code should be
```python
peak1 = psf.xy(fibers, zpeaks[0])
```
`PSF` subclasses aren't supposed to override `.xy`, but if they ever did then calling `PSF.xy(psf, fibers, zpeaks[0])` would be calling the wrong `.xy` calculation.  Same comment applies to the lines below.

i.e. you shouldn't need to `import PSF` at the top.

OTOH, if `psf.xy(fibers, zpeaks[0])` doesn't work, then we have a bug in that subclass and should fix it there.